### PR TITLE
highlight the difference from the original paper and RFC8312

### LINKED ISSUE #10: Highlight difference to paper
Would be nice to highlight what has changed from the original paper as that paper is still a good source to start with and understand cubic.

### PR DESCRIPTION
**Pull request by sangtaeha**
This includes the list of changes since the original paper. The section headings are similar to the issue logs at the end, so we may clean up at the end. Also, I didn't really install any tool to check this change (markdown and others). So please let me if this doesn't work. This fixes #10.

### CODE REVIEW

**Comment by sangtaeha** [comment]
@goelvidhi @larseggert It complains about the space (it seems I didn't remove the space when I split the sentence) and the reference to Lisong's paper. Can you fix this issue? If not, I will submit a new pull request.

**Comment by goelvidhi** [comment]
I am fixing the spaces. Will push soon.

**Comment by sangtaeha** [comment]
Thanks, @goelvidhi!

**Comment by goelvidhi** [comment]
I won't be able to push from your repo. Please find below the patch for spaces and tabs which you can apply using `git apply spaces.patch`
[spaces.patch.zip](https://github.com/NTAP/rfc8312bis/files/5641291/spaces.patch.zip)

**Comment by sangtaeha** [comment]
@goelvidhi Great! Let me fix it with your patch.

**Comment by goelvidhi** [comment]
I am not sure why the editor's copy check is failing. @larseggert can likely help here.

File: draft-eggert-tcpm-rfc8312bis.md [hunk 1]
```diff
@@ -802,6 +803,52 @@ Richard Scheffenegger and Alexander Zimmermann originally co-authored
 
 <!-- Anyone else to acknowledge? -->
 
+# History of changes since the original paper
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:806]: Please use title capitalization (also in your other headings)

**Comment by sangtaeha** [comment]
@goelvidhi Lisong's paper through DOI doesn't work. I think I may manually add that paper there like other papers.

File: draft-eggert-tcpm-rfc8312bis.md [hunk 2]
```diff
@@ -117,6 +117,7 @@ informative:
   CEHRX07:  DOI.10.1109/INFCOM.2007.111
   HRX08:    DOI.10.1145/1400097.1400105
   K03:      DOI.10.1145/956981.956989
+  SXEZ19:  DOI.10.5555/3323234.3323292
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:120]: This DOI doesn't seem to be correct (or exist?)

File: draft-eggert-tcpm-rfc8312bis.md [hunk 3]
```diff
@@ -802,6 +803,52 @@ Richard Scheffenegger and Alexander Zimmermann originally co-authored
 
 <!-- Anyone else to acknowledge? -->
 
+# History of changes since the original paper
+
+CUBIC has gone through a few changes since the initial release of its
+algorithm and implementation. Below we highlight the differences
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:809]: Reference the "initial relase"?

File: draft-eggert-tcpm-rfc8312bis.md [hunk 4]
```diff
@@ -802,6 +803,52 @@ Richard Scheffenegger and Alexander Zimmermann originally co-authored
...
+throughput using segments or bytes received instead of time t.
+When W_est > W_max, it sets alpha_aimd to 1.
+- The bugs reported in {{?SXEZ19}} are fixed. CUBIC sets W_cubic(t + RTT)
+as the target window size after the next RTT. However, this target may be
+too high, like even higher than 2 * cwnd (i.e., more aggressive than slow
+start) in the following cases: (1) RTT is extremely long; (2) after a long
+idle period; and (3) after a long application rate-limited period.
+To address this issue, CUBIC now has lower and upper bounds to ensure that
+the window increase rate is non-decreasing and is less than the increase
+rate of slow start.
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:827]: This seems redundant with the bullet lists in the next appendix. Maybe we add a placeholder here for now, and convert the bullet list into a section when we're done with the draft?

File: draft-eggert-tcpm-rfc8312bis.md [hunk 5]
```diff
@@ -802,6 +803,52 @@ Richard Scheffenegger and Alexander Zimmermann originally co-authored
...
+
+## RFC 8312
+
+- It changed the definition of beta_cubic constant and thus updated the
+pseudocode of CUBIC accordingly. For example, beta_cubic in the original
+paper was the window decrease constant while RFC8312 changed it to CUBIC
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:833]:
```suggestion
paper was the window decrease constant while {{!RFC8312}} changed it to CUBIC
```

File: draft-eggert-tcpm-rfc8312bis.md [hunk 6]
```diff
@@ -802,6 +803,52 @@ Richard Scheffenegger and Alexander Zimmermann originally co-authored
...
+multiplication decrease factor. With this change, the current congestion
+window size after a loss event is beta_cubic * W_max while it was
+(1-beta_cubic) * W_max in the original paper.
+- It used W_max while the original paper's pseudocode used W_last_max.
+- TCP friendly window is called W_est in RFC8312, whereas it was W_tcp in
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:838]:
```suggestion
- TCP friendly window is called W_est in {{!RFC8312}}, whereas it was W_tcp in
```

**Review by larseggert — CHANGES_REQUESTED** [review]

File: draft-eggert-tcpm-rfc8312bis.md (hunk 2 above)
Comment (sangtaeha) [reply draft-eggert-tcpm-rfc8312bis.md:120]: It looks like that is ACM's internal number. Let me manually add it. Thanks!

File: draft-eggert-tcpm-rfc8312bis.md (hunk 4 above)
Comment (sangtaeha) [reply draft-eggert-tcpm-rfc8312bis.md:827]: Yes, that makes sense.

**Review by larseggert — APPROVED** [review]

**Review by larseggert — CHANGES_REQUESTED** [review]
You need to wrap your lines and fix the other issues the checker found before this can be merged

**Comment by larseggert** [comment]
I'll fix the issue, hang on

**Comment by sangtaeha** [comment]
@larseggert I installed the tools and found that adding a new reference caused a build error. I exactly followed the format but somehow it complained about the new reference. At this time, I simply removed this new reference. I will experiment until I know how to add a new reference.

**Comment by larseggert** [comment]
The problem was subtle. For "automatic references" to RFCs and I-Ds, one uses `{{?foo}}` to indicate an informative reference to "foo", and `{{!bar}}` for a normative reference to "bar". And the `?` or `!` needs to be used consistently, otherwise the reference is included twice, causing an error.

Other references need to be included under `normative:` or `informative:`blocks at the beginning of the document. For those, one need not include `?` or `!` in the reference at all (because the status is known already), but if one includes `?` or `!` in the reference, it again needs to be done consistently.

**Comment by sangtaeha** [comment]
@larseggert Great. Thanks!

File: draft-eggert-tcpm-rfc8312bis.md (hunk 4 above)
Comment (larseggert) [reply draft-eggert-tcpm-rfc8312bis.md]: Could you add that placeholder?

**Review by lisongxu — APPROVED** [review]
Thanks, Sangtae!

**Review by goelvidhi — APPROVED** [review]

File: draft-eggert-tcpm-rfc8312bis.md [hunk 7]
```diff
@@ -805,6 +818,52 @@ Richard Scheffenegger and Alexander Zimmermann originally co-authored
...
+paper was the window decrease constant while {{!RFC8312}} changed it to CUBIC
+multiplication decrease factor. With this change, the current congestion
+window size after a loss event is beta_cubic * W_max while it was
+(1-beta_cubic) * W_max in the original paper.
+- It used W_max while the original paper's pseudocode used W_last_max.
+- TCP friendly window is called W_est in {{!RFC8312}}, whereas it was W_tcp in
+the original paper.
+- It included the discussion on safety features of CUBIC, such as CUBIC's
+fairness in small and high bandwidth-delay product (BDP) networks and the
+recommended CUBIC constant C.
```
Comment (larseggert) [inline draft-eggert-tcpm-rfc8312bis.md:857]: I think we should merge these two sections into the bullet lists below. Upon publication we can decide to convert the bullet lists to prose or remove them.

Comment (goelvidhi) [reply draft-eggert-tcpm-rfc8312bis.md:857]: @sangtaeha Lars and I discussed this offline. Perhaps we can something like:

Appedix A. Acknowledgments
Appendix B. Changes from RFC 8312
    B.1 We can combine the text under ## The Current Draft to the change log
    B.2 stays same
Appendix C. Changes from the original paper
We can move the text under ## RFC 8312 here.

**Review by goelvidhi — CHANGES_REQUESTED** [review]

**Comment by larseggert** [comment]
Would this be ready for a merge after the conflicts are resolved?

**Comment by sangtaeha** [comment]
@larseggert Yes, more likely. I rearranged them based on the feedback from you and @goelvidhi. After the merge, we can decide. The listed differences are more descriptive than the changelog, so we may either extend or shorten them.

**Comment by larseggert** [comment]
OK. I will merge once you resolve the conflicts to the `main` branch.

**Review by larseggert — APPROVED** [review]

**Comment by sangtaeha** [comment]
@larseggert I fixed it. Thanks!

**Pull request merged**